internal/controller/runtime/fsm/runtime_fsm_initialize_runtime_bootstrapper.go

```diff
@@ -0,0 +1,92 @@
+package fsm
+
+import (
+	"context"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+	"github.com/kyma-project/infrastructure-manager/internal/log_level"
+	"github.com/kyma-project/infrastructure-manager/internal/rtbootstrapper"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	msgStatusCheckFailed      = "Runtime bootstrapper status check failed"
+	msgInstallationFailed     = "Runtime bootstrapper installation failed"
+	msgInstallationInProgress = "Runtime bootstrapper installation in progress"
+	msgInstallationCompleted  = "Runtime bootstrapper installation completed"
+)
+
+func sFnInitializeRuntimeBootstrapper(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	if !m.RuntimeBootstrapperEnabled || m.RuntimeBootstrapperInstaller == nil {
+		m.log.V(log_level.DEBUG).Info("Runtime bootstrapper installation is disabled")
+		return switchState(sFnFinalizeRegistryCache)
+	}
+
+	status, err := m.RuntimeBootstrapperInstaller.Status(ctx, s.instance.Name)
+	if err != nil {
+		m.log.Error(err, "Failed to get runtime bootstrapper installation status")
+		s.instance.UpdateStatePending(
+			imv1.ConditionTypeRuntimeBootstrapperReady,
+			imv1.ConditionReasonRuntimeBootstrapperStatusUnknown,
+			"False",
+			msgStatusCheckFailed,
+		)
+		return updateStatusAndRequeue()
+	}
+
+	switch status {
+	case rtbootstrapper.StatusNotStarted:
+		{
+			err := m.RuntimeBootstrapperInstaller.Install(ctx, s.instance.Name)
+			if err != nil {
+				m.log.Error(err, "Failed to start runtime bootstrapper installation")
+				s.instance.UpdateStatePending(
+					imv1.ConditionTypeRuntimeBootstrapperReady,
+					imv1.ConditionReasonRuntimeBootstrapperInstallationFailed,
+					"False",
+					msgInstallationFailed,
+				)
+			} else {
+				s.instance.UpdateStatePending(
+					imv1.ConditionTypeRuntimeBootstrapperReady,
+					imv1.ConditionReasonRuntimeBootstrapperInstallationInProgress,
+					"False",
+					msgInstallationInProgress,
```

Comment:
Could you also add a unit test for this logic? It seems not to be covered by the unit test.